src/material-experimental/mdc-helpers/_mdc-helpers.scss

```diff
@@ -286,26 +321,46 @@ $mat-typography-level-mappings: (
   }
 }
 
-/// Mixin that sets the color of the focus indicators.
+/// Mixin that customizes focus indicators. This mixin may be included
+/// multiple times throughout your app.
 ///
-/// @param {color|map} $theme-or-color
-///   If theme, focus indicators are set to the primary color of the theme. If
-///   color, focus indicators are set to that color.
+/// @param {map} $config
+///   If given, customizes focus indicators with the configuration. The
+///   configuration map accepts border-style, border-color, border-width,
+///   border-radius, and margin. See the SassDoc annotation for
+///   mat-strong-focus-indicator for more on each parameter.
 ///
-/// @example
-///   .demo-dark-theme {
-///     @include mat-mdc-strong-focus-indicators-theme($dark-theme-map);
-///   }
+/// @param {boolean} $cascade
+///   If true, the focus indicator customizations will apply to the
+///   entire element subtree. Otherwise, they only apply to the selector
+///   hosting the mixin. Defaults to true.
 ///
 /// @example
-///   .demo-red-theme {
-///     @include mat-mdc-strong-focus-indicators-theme(#f00);
+///   .my-component {
+///     @include mat-mdc-strong-focus-indicator-customize((
+///       border-radius: 50%,
+///       color: red,
+///     ));
 ///   }
-@mixin mat-mdc-strong-focus-indicators-theme($theme-or-color) {
-  .mat-mdc-focus-indicator::before {
-    border-color: if(
-      type-of($theme-or-color) == 'map',
-      mat-color(map_get($theme-or-color, primary)),
-      $theme-or-color);
+@mixin mat-mdc-strong-focus-indicator-customize($config: (), $cascade: true) {
+  $config-keys: (
+    border-color,
+    border-style,
+    border-width,
+    border-radius,
+    margin,
+  );
+  $selector: if($cascade,
+      '.mat-mdc-focus-indicator::before',
```

Comment:
Does this work for things like `mat-button`? If someone tries to do:
```scss
.mat-button {
  @include mat-mdc-strong-focus-indicator-customize((border-color: red));
}
```

That would result in CSS like this:
```scss
.mat-button .mat-mdc-focus-indicator::before {
  border-color: red;
}
```

Which I believe would not work. They'd have to specify `$prevent-cascade: true` to make it work, but I think it would be confusing to users why that's the case

Reply:
Yeah that's a bummer. Does renaming the parameter help? e.g. `$target-self`, `$apply-to-self`, ...?

Wondering what alternatives you have in mind. One idea is that we could apply the customization styles to both the host selector as well as all children:

```
@mixin mat-strong-focus-indicator-customize() {
   &.mat-focus-indicator::before,
   .mat-focus-indicator::before {
     ...
   }
}
```

Another idea is to have two separate mixins instead of the `$cascade` flag.

Reply:
Maybe the `&.,.` thing, that sounds like it should give the expected behavior

Reply:
I think this hits on my original want to do separate mixins for each component, since it's difficult to do one thing that works across the board.

Reply:
The `&.,.` thing is not ideal because we serve twice as many styles per include, but at least we cover both scenarios without a separate parameter. The downside of having two mixins is that developers need to know when to use one over the other (like you said, it's not always obvious).

If we do separate mixins for each component, then we'll need an additional customize mixin that does cascade. Suppose I have a toolbar with a dark blue background and I want all focus indicators within to be `lightblue`. I want to write:

```
.dark-blue {
  @include mat-strong-focus-indicator-customize((border-color: lightblue));
```

If we just have a bunch of component-specific mixins, then there is no way to write the above.

Reply:
How about `mat-strong-focus-indicator-on-element` and `mat-strong-focus-indicator-on-descendants` ?